src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,302 @@
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Validation;
+using System.Net;
+using static NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationTask;
+using System.IO;
+
+namespace NzbDrone.Core.Download.Clients.DownloadStation
+{
+    public class DownloadStation : TorrentClientBase<DownloadStationSettings>
+    {
+        protected readonly IDownloadStationProxy _proxy;
+        protected static readonly Dictionary<DownloadStationTaskStatus, DownloadItemStatus> StatusTable = new Dictionary<DownloadStationTaskStatus, DownloadItemStatus>();
+
+        static DownloadStation()
+        {
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Waiting, DownloadItemStatus.Queued);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Error, DownloadItemStatus.Failed);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Paused, DownloadItemStatus.Paused);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Seeding, DownloadItemStatus.Completed);
+        }
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                     ITorrentFileInfoReader torrentFileInfoReader,
+                     IHttpClient httpClient,
+                     IConfigService configService,
+                     IDiskProvider diskProvider,
+                     IRemotePathMappingService remotePathMappingService,
+                     Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            this._proxy = proxy;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "Download Station";
+            }
+        }
+
+        protected virtual OsPath GetOutputPath(OsPath outputPath, DownloadStationTask torrent)
+        {
+            return outputPath + torrent.Title;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            IEnumerable<DownloadStationTask> torrents;
+
+            try
+            {
+                torrents = _proxy.GetItems(Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Error(ex, ex.Message);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var items = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var outputPath = new OsPath(torrent.Additional.Detail["destination"]);
+
+                if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+                {
+                    if (!Settings.TvDirectory.Equals(outputPath.FullPath))
```

Comment:
Sure.
Category is intended to filter the list of downloads provided by the download client. Many download clients have inherent support for categories (sometimes called labels) and support filtering on the api calls, which drastically improves performance if the client has hundreds of torrents.
Clients like Transmission don't support categories, so we have resorted to a workaround: We take transmission's default download dir and tack on the category name as a subdirectory.
That way we can filter out the api results so we don't try parse downloads that were added directly to transmission by the user or another app.
Sometimes ppl want to 'override' where the download is placed, instead of the default download dir.
Deluge (which supports labels) supports setting a download folder per label. But it's one of the few clients that support that.
So the TvDirectory option is intended to 'override' the default download directory in the torrent client.
Thus TvDirectory is mutually exclusive with Category, coz DownloadDIr=TvDirectory OR DownloadDir=DefaultDownloadDir+Category (OR DownloadDir=DefaultDownloadDir if none are specified).

The download directory can be seen as a 'base' directory.
Torrents with just one file are often placed directly in that directory. But Torrents with multiple files are placed in a new subdirectory (usually the Torrent Name). Sonarr has no control over this, torrent clients simply work that way.
You'll have to handle that properly in the GetOutputPath function.

What markus warned about is that you cannot plainly string compare the TvDirectory (the 'base' directory) with the OutputPath, coz it's a parent-child relation. There are helper functions to deal with that.